src/Idris/Doc/Keywords.idr

```diff
@@ -505,31 +505,71 @@ withabstraction = vcat $
     ```
     """]
 
-letbinding : Doc IdrisDocAnn
-letbinding = vcat $
+-- to keep docs consistent between `:doc let` and `:doc (:=)`
+letBinding : String
+letBinding =
+    """
+    * Let bindings are defined using the `:=` syntax and can be used to bind the
+    result of intermediate computations. A type annotation can be added but is
+    not required, for example:
+    ```idris
+    power4 : Nat -> Nat
+    power4 n = let square := n * n
+               in square * square
+
+    power4T : Nat -> Nat
+    power4T n = let square : Nat := n * n
+                in square * square
+    ```
+    Without the type annotation, `:=` behaves the same as `=` and gets
+    elaborated to the lambda expression `(\square => square * square) (n * n)`.
+
+    With the type annotation, `square` is the value of a computation, not its
+    generalisation, and so will only be computed once instead of potentially on
+    every reference.
```

Comment:
I don't understand these paragraphs

Reply:
I've attempted a clarification in the latest commit. Does that help?  : )

Reply:
It makes sense to me. I'd be inclined to ship this, as it's better than the current help text. But it's not up to me.

One thing I would like more clarification on is how the two forms might affect type inference. I.e. certain things work with one form or the other, and that can be quite surprising. I still don't have a good intuition for this.